lib/agenda/locklimit.js

```diff
@@ -3,11 +3,13 @@ const debug = require('debug')('agenda:locklimit');
 
 /**
  * Set the default amount jobs that are allowed to be locked at one time (GLOBAL)
- * NOTE: Is this different than max concurrency?
+ * @name Agenda#locklimit
+ * @function
  * @param {Number} num Lock limit
  * @returns {exports} agenda instance
  */
 module.exports = function(num) {
+   // @NOTE: Is this different than max concurrency?
```

Comment:
Yup!

This seems to be the single most confusing aspect of agenda (looking through issues etc.)

Agenda will lock jobs 1 by one, which means it will set `lockedAt` in mongoDB, create an instance of the `Job` class and cache it into the `_lockedJobs` array. This defaults to having no limit, but can be managed using `lockLimit`

Agenda will also pull jobs from `_lockedJobs` and into `_runningJobs`. These jobs are actively being worked on by user code, and this is limited by `maxConcurrency`.

Reply:
Definitely good material for docs. 👆  Wanna PR? :-)